packages/react-data-grid/src/CellAction.tsx

```diff
@@ -0,0 +1,49 @@
+import classNames from 'classnames';
+import React, { useState, ReactNode } from 'react';
+
+interface Action {
+  text: ReactNode;
+  callback(): void;
+}
+
+export interface CellActionProps {
+  icon: ReactNode;
+  actions?: Action[];
+  callback?(): void;
+  isFirst: boolean;
+}
+
+export default function CellAction({ icon, actions, callback, isFirst }: CellActionProps) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const cellActionClasses = classNames('rdg-cell-action', {
+    'rdg-cell-action-last': isFirst
+  });
+  const actionButtonClasses = classNames('rdg-cell-action-button', {
+    'rdg-cell-action-button-toggled': isOpen
+  });
+
+  function onActionIconClick() {
+    if (typeof callback === 'function') {
+      callback();
+    }
+
+    if (Array.isArray(actions) && actions.length > 0) {
+      setIsOpen(!isOpen);
+    }
+  }
+
+  return (
+    <div className={cellActionClasses} onMouseLeave={() => setIsOpen(false)}>
+      <div className={actionButtonClasses} onClick={onActionIconClick}>
+        {icon}
+      </div>
+      {
+        Array.isArray(actions) && actions.length && isOpen && (
```

Comment:
I think we should just trust our types. Here and above as well. WDYT?

```jsx
isOpen && actions.length > 0 && (
  ...
)
```

Reply:
I would like to say we should keep it and clean step by step. The type is safe inside this file, but the `Cell.js` is still not `tsx` yet

Reply:
Let's hope we won't forget to clean it then. 🤞

Reply:
`actions` is not a required field so we cannot change it

Reply:
https://github.com/adazzle/react-data-grid/blob/next/examples/scripts/example27-cell-actions.js#L134

Reply:
Right, but `Array.isArray` is still overkill probably.